### DISCUSSION TITLE
Async Notification is not processed

### DISCUSSION THREAD

**Discussion by karthik-tarento** [Sunbird-Lern/Community#44] [Issues]
Hello,

We are using sunbird-notification-service - release-4.4.0 version
Our certificate generator flink job is generating a request to send email when the user completes the course. The same is received in the sunbird-notification-service (via learner service) and initiating the send email functionality. 
This send email is not sync -- so, sunbird-notification-service writes the event into "dev.lms.notification" kafka topic. 
But we are not able to identify which service is consuming the messages from this topic. 
Please provide the details for the same.

Regards,
Karthik

**Comment by rhwarrier** [comment]
@reshmi-nair  can you please take a look ?

**Reply by reshmi-nair** [reply]
@karthik-tarento LMS jobs repo has a samza job which reads this kafka messages and sends the notification.

https://github.com/project-sunbird/sunbird-lms-jobs/

**Reply by karthik-tarento** [reply]
@reshmi-nair - thanks for the info.. will take a look at this code.

**Reply by karthik-tarento** [reply]
@reshmi-nair - is this lms job re-written as Flink job ? if so, please share the flink job repo link.

**Reply by prasadmoka** [reply]
@karthik-tarento please find the below link for the notification flink job:
https://github.com/Sunbird-Lern/data-pipeline/tree/release-5.0.0/notification/notification-job

**Comment by juhiagl8** [comment]
@reshmi-nair when we fire a notification kafka topic with the name of the vm template, will the notification flink job utilise this notification sdk and  convert the data wrt to the vm template?

**Reply by reshmi-nair** [reply]
@juhiagl8 Notification job sends the notification with the content that is received. It does not resolve the template or values in the content. So you need to sent the actual content instead of template name.
